pkg/arvo/lib/btc-node-json.hoon

```diff
@@ -0,0 +1,3129 @@
+    ::
+    ++  base58-to-cord
+      |=  b=@uc
+      ^-  @t
+      ::  Removes leading 0c
+      ::
+      (rsh 3 2 (scot %uc b))
+    ::  %hex-to-cord: parses hexadecimal to cords without dots and 0x
+    ::
+    ++  hex-to-cord
+      |=  h=@ux
+      ^-  @t
+      %-  crip
+      =-  ((x-co:co (mul 2 p)) q)
+      (as-octs:mimes:html h)
+    ::  %addr-type-validator: parses BTC addresses (legacy and p2sh) to @uc
+    ::
+    ::    bech32/segwit addressed are tagged separately.
+    ::
+    ++  addr-type-validator
+      |=  addr=@t
+      ^-  ?(@uc [%bech32 @t])
+      =,  dejs
+      ?~  (rush addr fim:ag)
+        ::  TODO: fim:ag doesn't parse %bech32 addresses
+        ::
+        bech32+addr
+      `@uc`(rash addr fim:ag)
+    ::  %to-hex: parses hexadecimal to @ux with separator dots and 0x
+    ::
+    ++  to-hex
+      |=  h=@t
+      ^-  @ux
+      ?:  =('' h)  0x0
+      ::  Add leading 00
+      ::
+      =+  (lsh 3 2 h)
+      ::  Group by 4-size block
+      ::
+      =+  (rsh 3 2 -)
+      ::  Parse hex to atom
+      ::
+      `@ux`(rash - hex)
+    ::
+    ++  ip-port-to-cord
+      |=  [ip=@if port=@ud]
+      ^-  cord
+      =/  a=tape  (scow %if ip)
+      =/  b=json  (numb:enjs:format port)
+      ?>  ?=([%n *] b)
+      ?~  a  ''
+      (cat 3 (cat 3 (crip t.a) ':') p.b)
+    ::
+    ++  to-wall
+      |=  =tape
+      ^-  wall
+      %+  roll  (flop tape)
+      |=  [char=@tD =wall]
+      ?~  wall
+        [[char ~] ~]
+      ?:  =('\0a' char)
+        [~ wall]
+      [[char i.wall] t.wall]
+    ::
+    ++  json-parser
+      |%
+      ::  %vin:json-parser
+      ::
+      ::    Used in:
+      ::    - $raw-transaction:json-parser
+      ::    - %decode-psbt
+      ::    - %decode-raw-transaction
+      ::
+      ++  vin
+        =-  (ar (ou -))
+        :~  ['txid' (uf ~ (mu (cu to-hex so)))]
+          ::
+            ['vout' (uf ~ (mu ni))]
+          ::
+            :-  'scriptSig'
+            =-  (uf ~ (mu (ot -)))
+            :~  ['asm' so]
+              ::
+                ['hex' (cu to-hex so)]
+            ==
+          ::
+            :-  'txinwitness'
+            =-  (uf ~ (mu -))
+            (ar (cu to-hex so))
+          ::
+            ['sequence' (un ni)]
+        ==
+      ::  %vout:json-parser
+      ::
+      ::    Used in:
+      ::    - $raw-transaction:json-parser
+      ::    - %decode-psbt
+      ::    - %decode-raw-transaction
+      ::
+      ++  vout
+        =-  (ar (ot -))
+        :~  ['value' no]
+          ::
+            ['n' ni]
+          ::
+            :-  'scriptPubKey'
+            %-  ou
+            :~  ['asm' (un so)]
+              ::
+                ['hex' (un (cu to-hex so))]
+              ::
+                ['req-sigs' (uf ~ (mu ni))]
+              ::
+                ['type' (un so)]
+              ::
+                :-  'addresses'
+                =-  (uf ~ (mu -))
+                (ar (cu addr-type-validator so))
+        ==  ==
```

Comment:
For cases like this, it's fine to keep the simpler ones grouped together, like so:

```hoon
:~  ['asm' (un so)]
    ['hex' (un (cu to-hex so))]
    ['req-sigs' (uf ~ (mu ni))]
    ['type' (un so)]
  ::
    :-  'addresses'
    =-  (uf ~ (mu -))
    (ar (cu addr-type-validator so))
==
```

This way, you cleanly delineate the complex codeblocks, but don't stretch out the simple ones unnecessarily.